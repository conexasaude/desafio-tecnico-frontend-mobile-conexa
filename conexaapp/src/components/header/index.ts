import {Button} from './button/view';
import {Greetings} from './greetings/view';
import {HStack} from './horizontal-stack/view';
import {Root} from './root/view';
import {Title} from './title/view';
import {VStack} from './vertical-stack/view';

export const Header = {
  Root: Root,
  Button: Button,
  Greetings: Greetings,
  VStack: VStack,
  HStack: HStack,
  Title: Title,
};
