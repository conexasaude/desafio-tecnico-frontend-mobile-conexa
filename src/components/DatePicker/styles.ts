import styled from '@emotion/native';

export const Wrapper = styled.View`
  flex: 1;
`;

export const Container = styled.TouchableOpacity(({ theme }) => ({
  height: 52,
  borderColor: theme.colors.primary,
  borderWidth: theme.effects.border.width.tl,
  borderRadius: theme.effects.border.radius.md,
  padding: theme.effects.spacing.vs,
  justifyContent: 'center',
  marginTop: theme.effects.spacing.md,
}));

export const Row = styled.View({
  flex: 1,
  flexDirection: 'row',
  alignItems: 'center',
});

export const Text = styled.Text(({ theme }) => ({
  flex: 1,
  fontFamily: theme.fonts.family.semiBold,
  fontSize: theme.fonts.size.lg,
  color: theme.colors.gray[900],
}));

export const Label = styled.Text(({ theme }) => ({
  position: 'absolute',
  top: -14,
  left: 16,
  fontFamily: theme.fonts.family.bold,
  fontSize: theme.fonts.size.md,
  color: theme.colors.primary,
  backgroundColor: theme.colors.white,
  padding: theme.effects.spacing.nn,
}));

export const Error = styled.Text(({ theme }) => ({
  fontFamily: theme.fonts.family.semiBold,
  fontSize: theme.fonts.size.md,
  color: theme.colors.status.error,
  marginTop: theme.effects.spacing.nn,
}));
