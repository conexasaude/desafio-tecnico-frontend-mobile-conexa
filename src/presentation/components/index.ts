export * from './buttons';
export * from './heading';
export * from './icons';
export * from './inputs';
