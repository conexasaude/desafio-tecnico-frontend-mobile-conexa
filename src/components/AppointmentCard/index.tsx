import { GestureResponderEvent, Pressable } from 'react-native';
import { Container, Pacient, AppointmentDate, Field, Title, Header } from './styles';
import { MaterialCommunityIcons } from '@expo/vector-icons'; 
import formatsDate from '../../utils/formatsDate';

interface AppointmentCardProps {
    id: string
    onPress: (event: GestureResponderEvent) => void
    pacient: string
    date: string
    observation: string
}

export default function AppointmentCard({ id, date, observation, onPress, pacient }: AppointmentCardProps) {

  return (
    <Container id={id} onPress={onPress}>
        <Header>
            <Title>Consulta #{id}</Title>
            <AppointmentDate>{formatsDate(date, 'day-month-year')}</AppointmentDate>
        </Header>
        <Field>
            <MaterialCommunityIcons name="account" size={20} color="#FFF" />
            <Pacient>{pacient}</Pacient>
        </Field>
        <Field>
            <MaterialCommunityIcons name="clock" size={20} color="#FFF" />
            <AppointmentDate>{formatsDate(date, 'time')}</AppointmentDate>
        </Field>
    </Container>
  );
}
