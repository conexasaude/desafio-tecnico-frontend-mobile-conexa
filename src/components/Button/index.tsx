import { ButtonContainer, ButtonText } from './styles';

interface ButtonProps {
  onPress: () => Promise<void>
  label: string
  disabled?: boolean
  testID?: string
}

export default function Login({ label, onPress, disabled, testID }: ButtonProps) {

  return (
    <ButtonContainer testID={testID} onPress={onPress} disabled={disabled}>
      <ButtonText>{label}</ButtonText>
    </ButtonContainer>
  );
}