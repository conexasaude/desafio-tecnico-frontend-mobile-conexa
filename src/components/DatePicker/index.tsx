import { GestureResponderEvent, Pressable, Text } from 'react-native';
import { Label, Placeholder, Value, ValueContainer } from './styles';

interface DatePickerProps {
    label: string
    value: Date | null
    onPress: ((event: GestureResponderEvent) => void) | null | undefined
}

export default function DatePicker({ label, value, onPress }: DatePickerProps) {
    function valueFormat(date: Date) {
        const formattedDate = new Intl.DateTimeFormat('pt-BR', {
			year: 'numeric',
			month: '2-digit',
			day: '2-digit',
			hour: '2-digit',
			minute: '2-digit',
		}).format(date);

        return formattedDate
    }
    return (
        <Pressable onPress={onPress}>
            <Label>
                {label}
                <Text style={{ color: 'red' }}> *</Text>
            </Label>
            
            <ValueContainer>
                {value ? <Value>{valueFormat(value)}</Value> : <Placeholder>Selecione uma data</Placeholder> }
            </ValueContainer>
        </Pressable>
    )
}