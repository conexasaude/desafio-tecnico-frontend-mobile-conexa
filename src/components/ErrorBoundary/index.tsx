import React from 'react';
import { useTheme } from '@emotion/react';
import Icon from 'react-native-vector-icons/MaterialCommunityIcons';
import * as Styled from './styles';

interface ErrorBoundaryProps {
  tryAgain: () => void;
}

export function ErrorBoundary({ tryAgain }: ErrorBoundaryProps) {
  const { colors } = useTheme();

  return (
    <Styled.Container testID="error-boundary">
      <Icon
        name="alert-circle-outline"
        color={colors.status.warning}
        size={64}
      />
      <Styled.Text>
        {'Desculpe, ocorreu um problema. Por favor, tente novamente.'}
      </Styled.Text>

      <Styled.ButtonTryAgain onPress={tryAgain} title="Tentar Novamente" />
    </Styled.Container>
  );
}
