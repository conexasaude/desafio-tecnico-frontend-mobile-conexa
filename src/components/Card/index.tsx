import React from 'react';
import * as Styled from './styles';
import { TouchableOpacityProps } from 'react-native';
import { ButtonProps } from '@/components/Button';

interface CardProps extends TouchableOpacityProps {
  title: string;
  description: string;
  button: ButtonProps;
}

export function Card({ title, description, button, ...rest }: CardProps) {
  return (
    <Styled.Wrapper
      testID="card"
      activeOpacity={0.8}
      onPress={button.onPress}
      {...rest}>
      <Styled.Row>
        <Styled.Column>
          <Styled.Title numberOfLines={1}>{title}</Styled.Title>
          <Styled.Description numberOfLines={2}>
            {description}
          </Styled.Description>
        </Styled.Column>

        <Styled.SeeMore
          testID="card-button"
          variant="outlined"
          activeOpacity={0.4}
          {...button}
        />
      </Styled.Row>
    </Styled.Wrapper>
  );
}
