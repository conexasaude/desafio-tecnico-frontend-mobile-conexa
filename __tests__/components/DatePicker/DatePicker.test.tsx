import React from 'react';
import {
  renderWithThemeProvider,
  matchSnapshotWithProvider,
} from '@/lib/functions/Testing';

import { DatePicker } from '@/components/DatePicker';
import { fireEvent } from '@testing-library/react-native';

describe('DatePicker', () => {
  test('should render correctly', () => {
    renderWithThemeProvider(<DatePicker label="label" />);
  });

  test('should matches snapshot', () => {
    matchSnapshotWithProvider(<DatePicker label="label" />);
  });

  test('should open the date picker modal on press', async () => {
    const { getByTestId, queryByTestId } = renderWithThemeProvider(
      <DatePicker label="label" />,
    );
    const datePicker = getByTestId('date-picker');
    const modal = queryByTestId('date-picker-modal');

    expect(modal).toBeNull();

    fireEvent.press(datePicker);

    const modalAfterPress = getByTestId('date-picker-modal');
    expect(modalAfterPress).toBeTruthy();
  });

  test('should be called onChange when confirm a date', () => {
    const mockOnChange = jest.fn();

    const { getByTestId, getByText } = renderWithThemeProvider(
      <DatePicker onChange={mockOnChange} label="label" />,
    );

    const datePicker = getByTestId('date-picker');
    fireEvent.press(datePicker);

    const confirmButton = getByText('confirm');
    fireEvent.press(confirmButton);

    expect(mockOnChange).toBeCalled();
  });

  test('should render placeholder', () => {
    const { getByText } = renderWithThemeProvider(
      <DatePicker placeholder="placeholder" label="label" />,
    );

    expect(getByText('placeholder')).toBeDefined();
  });

  test('should format label correclty with mode time', () => {
    let date = '';

    const { getByTestId, getByText } = renderWithThemeProvider(
      <DatePicker
        mode="time"
        onChange={newDate => (date = newDate)}
        label="label"
      />,
    );

    const datePicker = getByTestId('date-picker');
    fireEvent.press(datePicker);

    const confirmButton = getByText('confirm');
    fireEvent.press(confirmButton);

    expect(getByText(date)).toBeDefined();
  });

  test('should format label correclty with mode datetime', () => {
    let date = '';

    const { getByTestId, getByText } = renderWithThemeProvider(
      <DatePicker
        mode="datetime"
        onChange={newDate => (date = newDate)}
        label="label"
      />,
    );

    const datePicker = getByTestId('date-picker');
    fireEvent.press(datePicker);

    const confirmButton = getByText('confirm');
    fireEvent.press(confirmButton);

    expect(getByText(date)).toBeDefined();
  });

  test('should not update label if press cancel', () => {
    const { getByTestId, getByText } = renderWithThemeProvider(
      <DatePicker label="label" />,
    );

    expect(getByText('Selecione...')).toBeDefined();

    const datePicker = getByTestId('date-picker');
    fireEvent.press(datePicker);

    const cancelButton = getByText('cancel');
    fireEvent.press(cancelButton);

    expect(getByText('Selecione...')).toBeDefined();
  });

  test('should render error text', () => {
    const { getByText } = renderWithThemeProvider(
      <DatePicker error="something went wrong" label="label" />,
    );

    expect(getByText('something went wrong')).toBeDefined();
  });
});
