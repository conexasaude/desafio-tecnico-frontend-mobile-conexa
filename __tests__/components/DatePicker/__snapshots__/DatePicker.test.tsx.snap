// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DatePicker should matches snapshot 1`] = `
[
  <View
    style={
      [
        {
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
        },
        {},
      ]
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "borderColor": "#2E50D4",
          "borderRadius": 8,
          "borderWidth": 2,
          "height": 52,
          "justifyContent": "center",
          "marginTop": 16,
          "opacity": 1,
          "padding": 8,
        }
      }
      testID="date-picker"
    >
      <Text
        style={
          [
            {
              "backgroundColor": "#fff",
              "color": "#2E50D4",
              "fontFamily": "Nunito-Bold",
              "fontSize": 14,
              "left": 16,
              "padding": 4,
              "position": "absolute",
              "top": -14,
            },
            undefined,
          ]
        }
      >
        label
      </Text>
      <View
        style={
          [
            {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "row",
            },
            undefined,
          ]
        }
      >
        <Text
          style={
            [
              {
                "color": "#333333",
                "flex": 1,
                "fontFamily": "Nunito-SemiBold",
                "fontSize": 16,
              },
              undefined,
            ]
          }
        >
          Selecione...
        </Text>
        <Text
          allowFontScaling={false}
          selectable={false}
          style={
            [
              {
                "color": "#2E50D4",
                "fontSize": 20,
              },
              undefined,
              {
                "fontFamily": "Material Design Icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              {},
            ]
          }
        >
          ó°ƒ­
        </Text>
      </View>
    </View>
  </View>,
  <Modal
    hardwareAccelerated={false}
    visible={false}
  />,
]
`;
